File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    answer: int = 0
    tangerine_count: list[tuple] = Counter(tangerine).most_common()
    
    for i in range(len(tangerine_count)):
        if k > 0:
            k -= tangerine_count[i][1]
            answer += 1
        else:
            break

    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> count = new HashMap<>();
        
        for (int size : tangerine) {
            count.put(
                size,
                count.getOrDefault(size, 0) + 1
            );
        }
        
        List<Integer> counts = new ArrayList<>(count.values());
        counts.sort(Collections.reverseOrder());
        for (int c : counts) {
            k -= c;
            answer++;
            if (k <= 0) {
                break;
            }
        }
        
        return answer;
    }
}